src/pgduckdb_hooks.cpp

```diff
@@ -52,7 +52,11 @@ ContainsDuckdbFunctions(Node *node, void *context) {
 
 	if (IsA(node, Query)) {
 		Query *query = (Query *)node;
-		return query_tree_walker(query, ContainsDuckdbFunctions, context, 0);
+#if PG_VERSION_NUM >= 160000
+	return query_tree_walker(query, ContainsDuckdbFunctions, context, 0);
+#else
+	return query_tree_walker(query, (bool (*)()) ((void *) ContainsDuckdbFunctions), context, 0);
```

Comment:
Side-note: We might want to create a helper function with a consistent calling convention around such functions, so that we don't need so many `#if PG_VERSION_NUM` statements throughout the codebase. I expect query_tree_walker could become one of those. For now I think this is fine though, since we only call it twice.

But to be clear, let's not do that in this PR.

Reply:
Sorry, I didn't think of a good implementation here, can you help me think of a more elegant implementation to replace this

Reply:
Thank you for your guidance.